### PR TITLE
Refactor code for generating balancer channel args.

### PR DESCRIPTION
**Pull request by markdroth**
While working on #14450, I noticed that the way we were constructing channel args for the balancer channel was sub-optimal, so I put together this PR to fix it.  There are two specific changes here:

- Eliminate duplication between grpclb_channel.cc and grpclb_channel_secure.cc by moving all arg modifications that are common to both into grpclb.cc.  Now grpclb_channel.cc is a no-op and grpclb_channel_secure.cc implements only the modifications that are specific to secure channels.  As part of this, rename `grpc_lb_policy_grpclb_build_lb_channel_args()` to `grpc_lb_policy_grpclb_modify_lb_channel_args()`

- Fixed handling of removing call credentials for the balancer channel.  We were previously doing this in `grpc_lb_policy_grpclb_create_lb_channel()` instead of in `grpc_lb_policy_grpclb_modify_lb_channel_args()`, which meant that we were only doing it when constructing the channel args used when initially creating the balancer channel, not when constructing the channel args injected via the fake resolver.  However, [`grpc_channel_args_union()` gives precedence to its first argument when a key is present in both arguments](https://github.com/grpc/grpc/blob/master/src/core/lib/channel/channel_args.h#L51), and [the fake resolver passes its injected arguments as the first argument](https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/fake/fake_resolver.cc#L124), which I believe means that we were never actually using the modified credentials in the balancer channel.  (This is only based on reading the code; we are not currently testing grpclb with a secure channel, so no test was catching this.  I've added a note about this to the existing TODO in grpclb_end2end_test.)  As part of this, eliminated the `grpc_lb_policy_grpclb_create_lb_channel()` function.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +120%    +462 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc           +462  +120%
      [NEW]    +815 grpc_lb_policy_grpclb_modify_lb_channel_args                                            +815  [NEW]
      +175%     +14 [Unmapped]                                                                               +14  +175%
      [NEW]      +5 balancer_name_cmp_fn                                                                      +5  [NEW]
      [NEW]      +5 destroy_balancer_name                                                                     +5  [NEW]

 -------------- SHRINKING                                                                            --------------
  -0.1%    -318 [None]                                                                                  -806  -0.0%
      -0.1%    -310 [Unmapped]                                                                              -798  -0.0%
      [DEL]     -32 grpc_lb_policy_grpclb_build_lb_channel_args(grpc_slice_hash_table*, grpc_core::FakeR     -32  [DEL]
      [DEL]      -8 grpc_lb_policy_grpclb_create_lb_channel(char const*, grpc_client_channel_factory*, g      -8  [DEL]
  -2.5%    -368 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                          -368  -2.5%
      [DEL]    -801 build_lb_channel_args(grpc_lb_addresses const*, grpc_core::FakeResolverResponseGener    -801  [DEL]
      [DEL]    -145 lb_call_data_shutdown(glb_lb_policy*) [clone .isra.4]                                   -145  [DEL]
      -4.2%     -18 [Unmapped]                                                                               -18  -4.2%
      -2.2%     -16 glb_update_locked                                                                        -16  -2.2%
      -0.7%      -8 glb_create                                                                                -8  -0.7%
      [DEL]      -5 balancer_name_cmp_fn                                                                      -5  [DEL]
      [DEL]      -5 destroy_balancer_name                                                                     -5  [DEL]

  -0.0%    -224 TOTAL                                                                                   -712  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +120%    +462 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc           +462  +120%
      [NEW]    +815 grpc_lb_policy_grpclb_modify_lb_channel_args                                            +815  [NEW]
      +175%     +14 [Unmapped]                                                                               +14  +175%
      [NEW]      +5 balancer_name_cmp_fn                                                                      +5  [NEW]
      [NEW]      +5 destroy_balancer_name                                                                     +5  [NEW]

 -------------- SHRINKING                                                                            --------------
  -0.1%    -318 [None]                                                                                  -806  -0.0%
      -0.1%    -310 [Unmapped]                                                                              -798  -0.0%
      [DEL]     -32 grpc_lb_policy_grpclb_build_lb_channel_args(grpc_slice_hash_table*, grpc_core::FakeR     -32  [DEL]
      [DEL]      -8 grpc_lb_policy_grpclb_create_lb_channel(char const*, grpc_client_channel_factory*, g      -8  [DEL]
  -2.5%    -368 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                          -368  -2.5%
      [DEL]    -801 build_lb_channel_args(grpc_lb_addresses const*, grpc_core::FakeResolverResponseGener    -801  [DEL]
      [DEL]    -145 lb_call_data_shutdown(glb_lb_policy*) [clone .isra.4]                                   -145  [DEL]
      -4.2%     -18 [Unmapped]                                                                               -18  -4.2%
      -2.2%     -16 glb_update_locked                                                                        -16  -2.2%
      -0.7%      -8 glb_create                                                                                -8  -0.7%
      [DEL]      -5 balancer_name_cmp_fn                                                                      -5  [DEL]
      [DEL]      -5 destroy_balancer_name                                                                     -5  [DEL]

  -0.0%    -224 TOTAL                                                                                   -712  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by markdroth** [comment]
Please hold off on reviewing this for a bit -- I'm going to merge #13870 first and then modify this PR accordingly.

**Comment by markdroth** [comment]
This is now ready for review.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +120%    +462 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb_channel_secure.cc           +462  +120%
      [NEW]    +815 grpc_lb_policy_grpclb_modify_lb_channel_args                                            +815  [NEW]
      +175%     +14 [Unmapped]                                                                               +14  +175%
      [NEW]      +5 balancer_name_cmp_fn                                                                      +5  [NEW]
      [NEW]      +5 destroy_balancer_name                                                                     +5  [NEW]

 -------------- SHRINKING                                                                            --------------
  -0.1%    -374 [None]                                                                                  -862  -0.0%
      -0.1%    -334 [Unmapped]                                                                              -854  -0.0%
      -2.8%     -32 [None]                                                                                     0  [ = ]
      [DEL]     -32 grpc_lb_policy_grpclb_build_lb_channel_args(grpc_slice_hash_table*, grpc_core::FakeR     -32  [DEL]
      [DEL]      -8 grpc_lb_policy_grpclb_create_lb_channel(char const*, grpc_client_channel_factory*, g      -8  [DEL]
  -2.0%    -304 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                          -304  -2.0%
     -21.9%    -289 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked                      -289 -21.9%
      -2.8%     -10 [Unmapped]                                                                               -10  -2.8%
      [DEL]      -5 grpc_core::(anonymous namespace)::BalancerNameCmp                                         -5  [DEL]

  -0.0%    -216 TOTAL                                                                                   -704  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by markdroth** [comment]
Known failures: #13612 #13385 #14475

File: test/cpp/end2end/grpclb_end2end_test.cc [hunk 1]
```diff
@@ -58,6 +58,8 @@
 // - Test handling of creation of faulty RR instance by having the LB return a
 //   serverlist with non-existent backends after having initially returned a
 //   valid one.
+// - test using secure credentials and make sure we don't send call
```
Comment (dgquintas) [inline test/cpp/end2end/grpclb_end2end_test.cc]: This is done in https://github.com/grpc/grpc/pull/14507

Comment (markdroth) [reply test/cpp/end2end/grpclb_end2end_test.cc]: Thanks.  I'll merge this as-is now and let you remove this TODO as part of your PR.

**Pull request merged**